src/main/xerus/monstercat/api/Player.kt

```diff
@@ -201,7 +213,25 @@ object Player : FadingHBox(true, targetHeight = 25) {
 				track.artists.map { artists.contains(it.name).to(3, 0) }.average() +
 						(track.titleRaw == title).toInt() + (track.artistsTitle == artists).to(10, 0)
 			}!!)
-			player?.setOnEndOfMedia { stopPlaying() }
+			if (Playlist.playlist.isEmpty()) {
+				player?.setOnEndOfMedia { stopPlaying() }
+			} else {
+				if (Playlist.random){
```

Comment:
You should put the ifs inside of the `player?.setOnEndOfMedia`, then they are checked when the next song has to be chosen, which is much more sensible.

Reply:
This is better :
```kotlin
player?.setOnEndOfMedia {
    if (Playlist.playlist.isEmpty()) stopPlaying()
    else {
        val s = if (Playlist.random) Playlist.nextRandom() else Playlist.next()
        if (s != null) play(s.title, s.artists) else stopPlaying()
    }
}